src/index.js

```diff
@@ -55,6 +56,25 @@ class FluentElement extends HTMLElement {
   }
 
   connectedCallback() {
+    this.dispatchEvent(
+      new CustomEvent("fluent-bundles-subscribe", {
+        bubbles: true,
+        target: this,
+      })
+    );
+    this.render();
+  }
+  disconnectedCallback() {
+    this.dispatchEvent(
+      new CustomEvent("fluent-bundles-unsubscribe", {
+        bubbles: true,
+        target: this,
+      })
+    );
+  }
+
+  set providerBundles(newBundles) {
+    this._providerBundles = CachedSyncIterable.from(newBundles);
```

Comment:
I'm thinking that this and `set bundles(newBundles) {` should both check to see if `newBundles` is anything other than an array, and if so don't use `CachedSyncIterable.from()` and instead convert `this._providerBundles` and `this._bundles` to `null`.

The later is a bug in my code I think.

Reply:
Done. I allowed any iterable instead of just arrays, throw a `fluent-web-error` if it's not iterable or null, and applied the same check to `FluentElement.bundle` too - sound okay?

Reply:
Awesome, thanks for all the work! Merging now and I'll deploy to npm a bit later tonight.